cli/src/action/mod.rs

```diff
@@ -98,11 +99,21 @@ fn chown(path: &Path, uid: u32, gid: u32) -> Result<(), CliError> {
 }
 
 /// Reads a private key from the given file name.
-fn read_private_key(file_name: &str) -> Result<String, CliError> {
+///
+/// # Arguments
+///
+/// * `file_name` - name or full path of the private key file
+/// * `print_file_name` - a printable version of the file name or path, to be used in
+///    reporting errors allowing the `file_name` to have a type that does not
+///    implement std::fmt::Display
+fn read_private_key<T: AsRef<Path> + Copy>(
+    file_name: T,
+    print_file_name: &str,
+) -> Result<String, CliError> {
     let mut file = File::open(file_name).map_err(|err| {
         CliError::EnvironmentError(format!(
             "Unable to open key file '{}': {}",
-            file_name,
+            print_file_name,
```

Comment:
This can probably just be file_name.display()

Reply:
Reverted this function to the original because it isn't needed in 'create_token' after moving key loading functionality to cylinder - fixed